functions/Set-DbaRgResourcePool.ps1

```diff
@@ -0,0 +1,229 @@
+    [CmdletBinding(SupportsShouldProcess, DefaultParameterSetName = "Default", ConfirmImpact = "Low")]
+    param (
+        [parameter(ValueFromPipeline, ValueFromPipelineByPropertyName)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$ResourcePool,
+        [ValidateSet("Internal", "External")]
+        [string]$Type = "Internal",
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumCpuPercentage,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumCpuPercentage,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(1, 100)]
+        [int]$CapCpuPercentage,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 100)]
+        [int]$MinimumMemoryPercentage,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [ValidateRange(1, 100)]
+        [int]$MaximumMemoryPercentage,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(0, 2147483647)]
+        [int]$MinimumIOPSPerVolume,
+        [Parameter(ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName = "Internal")]
+        [ValidateRange(1, 2147483647)]
+        [int]$MaximumIOPSPerVolume,
+        [Parameter(ParameterSetName = "External")]
+        [int]$MaximumProcesses,
+        [switch]$SkipReconfigure,
+        [Parameter(ValueFromPipeline)]
+        [object[]]$InputObject,
+        [switch]$EnableException
+    )
+
+    process {
+        if (-not $InputObject -and -not $ResourcePool) {
+            Stop-Function -Message "You must pipe in a resource pool or specify a ResourcePool."
+            return
+        }
+        if (-not $InputObject -and -not $SqlInstance) {
+            Stop-Function -Message "You must pipe in a resource pool or specify a SqlInstance."
+            return
+        }
+
+        if ($ResourcePool) {
+            $InputObject += Get-DbaRgResourcePool -SqlInstance $SqlInstance -Type $Type | Where-Object { $_.Name -in $ResourcePool }
+            if ($null -eq $InputObject) {
+                Stop-Function -Message "No resources pools matching '$ResourcePool' found." -Category ObjectNotFound -Target $existingResourcePool -Continue
+                return
+            }
+        }
+
+        if (($InputObject) -and ($PSBoundParameters.Keys -notcontains 'Type')) {
+            if ($InputObject -is [Microsoft.SqlServer.Management.Smo.ResourcePool]) {
+                $Type = "Internal"
+            } elseif ($InputObject -is [Microsoft.SqlServer.Management.Smo.ExternalResourcePool]) {
+                $Type = "External"
+            }
+        }
+
+        foreach ($instance in $SqlInstance) {
```

Comment:
Have you tested this with multiple objects piped in?

Reply:
Adding a multi-object pipe test for `Set-` and made the `Remove-` test use two piped objects versus the previous one.